web/components/TextBlock/Programs/Programs.tsx

```diff
@@ -15,19 +15,20 @@ import styles from './Programs.module.css';
 
 type ProgramsProps = {
   type: EntitySectionSelection;
+  programs?: Program[];
   allPrograms: Program[];
 };
 
 export const Programs = ({
-  value: { type, allPrograms },
+  value: { type, programs, allPrograms },
   index,
 }: PortableTextComponentProps<ProgramsProps>) => {
-  if (type === 'all') {
+  if (type === 'all' || type === 'highlighted') {
```

Comment:
Nit: This doesn't work properly in the following scenario:

1. (In Sanity Studio) Create a programs section with type 'highlighted' and select the Oslo program. Publish.
2. (In the app) Programs section now shows Oslo only, as expected.
3. (In Sanity Studio) Change the programs section type from 'highlighted' to 'all'. Publish.
4. (In the app) Programs section now still only shows Oslo

This is currently demonstrated on the /shared-sections-test page.

The studio doesn't clear out the `programs` array, it just hides it. Which might theoretically be a feature if you want to switch back from "all" to the old "highlighted" configuration later on, I suppose.

Suggestion part 1/2:
```suggestion
  if (type === 'all' || type === 'highlighted') {
    const selectedPrograms = type === 'all' ? allPrograms : programs;
```

Reply:
Interesting catch! What do you think about eb14b33 and its application in b5e4dbf? This was applied to the sections where it was straightforward to do so; a larger-scale unification is probably in order to streamline using this in some sort of `<EntitySection />` component.

Reply:
I think that looks OK, at least for now. I would think that if https://github.com/sanity-io/structured-content-2022/issues/91 is solved then it would no longer be relevant, as we would avoid fetching the irrelevant collection from Sanity in the first place.